core/src/main/java/org/apache/seata/core/protocol/detector/Http2Detector.java

```diff
@@ -50,17 +54,31 @@ public boolean detect(ByteBuf in) {
 
     @Override
     public ChannelHandler[] getHandlers() {
-        return new ChannelHandler[]{
-            Http2FrameCodecBuilder.forServer().build(),
+        return new ChannelHandler[] {Http2FrameCodecBuilder.forServer().build(),
             new Http2MultiplexHandler(new ChannelInitializer<Http2StreamChannel>() {
                 @Override
                 protected void initChannel(Http2StreamChannel ch) {
                     final ChannelPipeline p = ch.pipeline();
-                    p.addLast(new GrpcDecoder());
-                    p.addLast(new GrpcEncoder());
-                    p.addLast(serverHandlers);
+                    p.addLast(new ChannelInboundHandlerAdapter() {
```

Comment:
Should we not use an inner class here as we did before?

Reply:
done